plutus-use-cases/src/Plutus/Contracts/GameBabbage.hs

```diff
@@ -0,0 +1,252 @@
+{-# LANGUAGE AllowAmbiguousTypes        #-}
```

Comment:
How about moving the duplication between `Game` and `GameBabbage` into a separate module? It could be something like:

* create the `Plutus.Contract.Game.Types` module and move the GameParams, LockArgs, and UnlockArgs over there.
* create the `Plutus.Contract.Game.PlutusV1` module
* create the `Plutus.Contract.Game.PlutusV2` module

The reason I'm proposing `PlutusV1` and `PlutusV2` instead of `Alonzo` and `Babbage` respectively is because the tx that we are creating are *always* `Babbage`. The difference between the two examples are mostly the Plutus version and features of a specific era.

Reply:
I like the submodule idea, but I don't if Plutus version are better than era: new features comes with era and are supported by Plutus versions. The original script were a Plutus v2 script, that didn't take advantage of the Babbage feature... I like the era more, because one could look at the era specification and see what's new. But I'm gonna introduce the submodule (and document the meaning of the eras - ie. only use features up to the given era)

Reply:
You're right. Forgot that all plutus-use-cases where updated to PlutusV2. In that case, it does make sense.